api/jobs/handlers.py

```diff
@@ -474,18 +414,12 @@ def get_logs_html(self, _id):
 
         return
 
+    @require_admin
     def add_logs(self, _id):
-        """Add to a job's logs"""
-        if not self.superuser_request and not self.user_is_admin:
-            self.abort(403, 'Request requires admin')
-
-
         doc = self.request.json
 
-        try:
-            Job.get(_id)
-        except Exception: # pylint: disable=broad-except
-            raise APINotFoundException('Job not found')
+        j = Job.get(_id)
+        Queue.mutate(j, {}) # Unconditionally heartbeat
```

Comment:
The removal of the exception handling causes this endpoint to `500` when the job does not exist rather than `404`. IMO we should be moving towards a state where the API only `500`s on server error, not user/client error. You could switch the `Jobs.get` error to an `APINotFoundExeception` to avoid having to catch Exceptions in handler layers, like we do in container storage classes.

Reply:
I checked down the call stack, and `APINotFoundExeception` is what is thrown in this case. Unless I misread.

Reply:
You were right, I misread. I believe it used to be an Exception and I didn't look close enough.